analysers/trials/numberOfProcessInstances.py

```diff
@@ -23,10 +24,95 @@
     .set("spark.cassandra.connection.host", cassandraHost)
 sc = CassandraSparkContext(conf=conf)
 
-data = sc.cassandraTable(cassandraKeyspace, srcTable)\
-        .select("duration") \
+dataRDD = sc.cassandraTable(cassandraKeyspace, srcTable)\
+        .select("process_definition_id", "source_process_instance_id", "end_time", "start_time", "duration") \
         .where("trial_id=? AND experiment_id=?", trialID, experimentID) \
-        .map(lambda r: 1) \
+        .filter(lambda r: r["process_definition_id"] is not None) \
+        .cache()
+
+processes = dataRDD.map(lambda r: r["process_definition_id"]) \
+        .distinct() \
+        .collect()
+
+maxTime = None
+maxID = None
+for p in processes:
+    time = dataRDD.filter(lambda r: r["process_definition_id"] == p) \
+        .map(lambda r: (r["start_time"], r["source_process_instance_id"])) \
+        .sortByKey(1, 1) \
```

Comment:
You can expect to have the same ordering out of multiple `sortByKey` call, but I would suggest to also order by `source_process_instance_id` when the `start_time` is equal, and do it here and in all the computation below regarding data preprocessing.